Снова генки. Медсканер эдишн

4310V343k opened the pull request:
- [x] Генок видно на сканере только если у них есть линговский гендамаг
- [x] Передоз риеталина у генок делает ему рандомную внешку (уточнить, что именно)
- [x] Передоз риеталина у человеков дает сильный, поправимый только через хирургию урон по органам

resolves #6683 

<details>
<summary>Чейнджлог</summary>

```yml
🆑 remindme
balance: Генок видно на мед. сканерах только если у них есть линговский геномный дамаг.
rscadd: Передоз риеталина у генок вызывает геномный дамаг и случайные изменения внешности.
rscadd: Передоз риеталина у людей вызывает обильный урон по внутренним органам, излечимый только через хирургию.
/🆑
```

</details>

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

4310V343k committed:
1. biostructure is hidden, unless ling has genetic damage
also some feedback for the player

4310V343k committed:
pass consumed reagent amount to overdose handlers

4310V343k committed:
change geneticdamage to genome_damage to avoid confusion with organs

4310V343k committed:
3. ryetalyn OD changes ling's appearance

4310V343k committed:
correct clamp

4310V343k commented:
~Разрешаю обсирасть мой первый код за год~
@TobyThorne, расскажи пж поподробнее, что именно должно изменяться у генок под передозом?
И как в вашем билде лучше всего реализовать последнюю фичу?
~И что-то мне подсказывает, что это должны были быть либо два разных ишшуя, либо два разных ПРа~

kreeperHLC commented:
> ~Разрешаю обсирасть мой первый код за год~ @TobyThorne, расскажи пж поподробнее, что именно должно изменяться у генок под передозом? И как в вашем билде лучше всего реализовать последнюю фичу? ~И что-то мне подсказывает, что это должны были быть либо два разных ишшуя, либо два разных ПРа~

Так он в драфте же, чего тут смотреть

kreeperHLC commented on code/game/gamemodes/changeling/changeling_power.dm:
```diff
@@ -13,7 +13,7 @@
 	var/required_chems = 0 // Chemicals required to use the ability
 	var/required_dna = 0   // DNA required to use the ability
-	var/max_genetic_damage = 100 // Can't use the ability if genetic damage is higher than this
+	var/max_genome_damage = 100 // Can't use the ability if genetic damage is higher than this
```
```suggestion
	var/max_genome_damage = 100 // Can't use the ability if genome damage is higher than this
```

kreeperHLC commented on code/game/gamemodes/changeling/changeling_datum.dm:
```diff
@@ -81,6 +81,26 @@ var/global/list/possible_changeling_IDs = list("Alpha","Beta","Gamma","Delta","E
+	new_damage = clamp(new_damage, 0, 200)
+
+	if(new_damage == 0 && genome_damage > 0 && my_mob) // hide the biostructure if no gendamage
+		to_chat(my_mob, SPAN("changeling", "You feel your genomes have assembled. Your special biostructure cannot be easily seen now."))
```
Какое you, если генко это we. А ещё мне не нравится слово special тут

4310V343k replied:
Как-то не подумал. И везде так же

4310V343k replied:
```suggestion
		to_chat(my_mob, SPAN("changeling", "We feel our genomes have assembled. Our biostructure cannot be easily seen now."))
```

kreeperHLC commented on code/modules/reagents/Chemistry-Reagents.dm:
```diff
@@ -67,6 +63,10 @@
+	if(overdose && (location != CHEM_TOUCH))
+		var/overdose_threshold = overdose * (flags & IGNORE_MOB_SIZE? 1 : MOB_MEDIUM/M.mob_size)
+		if(volume > overdose_threshold)
+			overdose(M, alien, removed)
```
А чего тут происходит и почему вниз убрал можешь объяснить?

4310V343k replied:
Скорее всего проскочило мимо, пока коммитил. Возможно, была идея количеству дамага зависеть от количества реагентов при передозе, прям как и при обычном употреблении

kreeperHLC commented on code/modules/reagents/Chemistry-Reagents/medicine.dm:
```diff
@@ -359,6 +359,37 @@
+			if(CP.type == /datum/changeling_power/transform) // assume they must have a transform power
+				var/datum/changeling_power/transform/TR = CP
+
+				to_chat(M, SPAN_DANGER("You feel your genome is going bananas!"))
```
НАШ геном

4310V343k replied:
![i](https://user-images.githubusercontent.com/56449763/194788293-62f10f42-4d17-44c8-873a-7dd7a9d790ac.jpeg)
```suggestion
				to_chat(M, SPAN_DANGER("We feel our genome is going bananas!"))
```

4310V343k committed:
Apply suggestions from code review

Co-authored-by: goliath <kreeper3211225@gmail.com>

4310V343k committed:
Revert "pass consumed reagent amount to overdose handlers"

This reverts commit 35494eb08b01875ba78ace8e7da89fe16ee32948.

kreeperHLC commented:
Ну это, открываешь с Драфта иль что?

4310V343k commented:
> Ну это, открываешь с Драфта иль что?

ну так таски с ишшуя не сделаны полностью
если не нужен эффект от передоза риеталина у людей - так и скажите, но тогда всем просто будут вкалывать его по мете

kreeperHLC commented:
> > Ну это, открываешь с Драфта иль что?
> 
> ну так таски с ишшуя не сделаны полностью если не нужен эффект от передоза риеталина у людей - так и скажите, но тогда всем просто будут вкалывать его по мете

У тебя тут стейл уже как 23 дня. Ну ты решай что-то.

4310V343k commented:
> > > Ну это, открываешь с Драфта иль что?
> > 
> > 
> > ну так таски с ишшуя не сделаны полностью если не нужен эффект от передоза риеталина у людей - так и скажите, но тогда всем просто будут вкалывать его по мете
> 
> У тебя тут стейл уже как 23 дня. Ну ты решай что-то.

Я-то с драфта уберу, но тогда всем подряд по мете будут риеталин колоть

Tsurupeta reviewed (dismissed):
не по иссую

4310V343k committed:
4. ryetalyn OD causes massive internal damage to non-lings
technically, satisfies the issue. if the person is dead then the only way to fix dead organs is surgery. right?

4310V343k committed:
Merge remote-tracking branch 'upstream/dev' into lings!1

Doster-d commented:
Автор ты живой?!

4310V343k commented:
Нет. Я не оживу, пока мне не скажут что именно не по ишшую

Doster-d commented on code/modules/reagents/Chemistry-Reagents/medicine.dm:
```diff
@@ -359,6 +359,41 @@
+
+	// not a changeling. anti meta them to death
+	for(var/obj/item/organ/internal/I in M.internal_organs)
+		I.take_internal_damage(10, silent=TRUE)
```
Овердоз тикает каждуй тик, врач тупо не успеет вылечить, может поставим большой шанс на то, что войдет в фор?

Tsurupeta replied:
тут надо на метаболизм умножать, как это сделано почти везде с подгонкой значений до приемлемых, 10 дамага в тик это примерно 17 секунд жизни пациента. Мозг, вроде, тоже у нас внутренний орган.

Tsurupeta commented on code/game/gamemodes/changeling/changeling_datum.dm:
```diff
@@ -81,6 +81,26 @@ var/global/list/possible_changeling_IDs = list("Alpha","Beta","Gamma","Delta","E
+	if(new_damage == 0 && genome_damage > 0 && my_mob) // hide the biostructure if no gendamage
+		to_chat(my_mob, SPAN("changeling", "We feel our genomes have assembled. Our biostructure cannot be easily seen now."))
+		var/obj/item/organ/internal/biostructure/BIO = locate() in my_mob.contents
+		if(BIO)
+			BIO.hidden = TRUE
```
можно без капса пж?

4310V343k replied:
делал так же, как в проках рядом, но ок

Tsurupeta commented on code/game/gamemodes/changeling/changeling_datum.dm:
```diff
@@ -81,6 +81,26 @@ var/global/list/possible_changeling_IDs = list("Alpha","Beta","Gamma","Delta","E
+	if(new_damage > 0 && genome_damage == 0 && my_mob) // show the biostructure if we gain some gendamage
+		to_chat(my_mob, SPAN("changeling", "You feel your genomes start to disassemble. Your special biostructure can now be easily spotted."))
+		var/obj/item/organ/internal/biostructure/BIO = locate() in my_mob.contents
+		if(BIO)
+			BIO.hidden = FALSE
```
и тут

Tsurupeta commented on code/modules/organs/internal/_internal.dm:
```diff
@@ -12,6 +12,7 @@
 	var/min_bruised_damage = 10       // Damage before considered bruised
 	var/foreign = FALSE 			  // foreign organs shouldn't be removed or recreated on revive
 	var/override_species_icon = FALSE // Should we ignore species-specific icons?
+	var/hidden = FALSE				  // Should this organ be hidden on scanners?
```
там разве нет уже похожей системы?

4310V343k replied:
не нашел 🤔

Tsurupeta commented on code/modules/reagents/Chemistry-Reagents/medicine.dm:
```diff
@@ -359,6 +359,41 @@
 
+/datum/reagent/ryetalyn/overdose(mob/living/carbon/M, alien)
+	. = ..()
+	if(M.mind?.changeling && world.time - M.mind.changeling.last_transformation_at > 100)
+		// changelings get random appearance, since they don't have an initial one
+		for(var/datum/changeling_power/CP in M.mind.changeling.available_powers)
+			if(CP.type == /datum/changeling_power/transform) // assume they must have a transform power
```
магическое значение плюс какой-то странный фор

4310V343k committed:
code review

4310V343k committed:
let's lower the damage a bit

4310V343k committed:
Merge branch 'dev' into lings!1

NefrNerf commented:
Автор жив, ревувать можно? Идея то топ

4310V343k commented:
так хз вроде апрув уже стоит один

NefrNerf commented on code/game/gamemodes/changeling/changeling_datum.dm:
```diff
@@ -81,6 +81,26 @@ var/global/list/possible_changeling_IDs = list("Alpha","Beta","Gamma","Delta","E
+
+	if(new_damage == 0 && genome_damage > 0 && my_mob) // hide the biostructure if no gendamage
+		to_chat(my_mob, SPAN("changeling", "We feel our genomes have assembled. Our biostructure cannot be easily seen now."))
+		var/obj/item/organ/internal/biostructure/biostructure = locate() in my_mob.contents
```
Можешь объяснить зачем этот тут? У нас есть кейс когда будет аплаиться демедж но биоструктуры нет в генокраде?

Doster-d replied:
Видимо он хочет использовать это, когда у человека отсоединили биоструктуру, но не удалили из тела, видимо...

Doster-d replied:
Хотя правильное было бы использовать список внутренних органов, но может у автора есть свое оправдание использование локейта.

4310V343k replied:
копипаста iirc, я это кодил год назад

4310V343k replied:
> Видимо он хочет использовать это, когда у человека отсоединили биоструктуру, но не удалили из тела, видимо...

а че так можно?

Doster-d replied:
Ну да, из интерналов удаляется и остается в контентах моба

4310V343k replied:
звучит как что-то что не должно происходить

NefrNerf commented on code/game/gamemodes/changeling/changeling_datum.dm:
```diff
@@ -81,6 +81,26 @@ var/global/list/possible_changeling_IDs = list("Alpha","Beta","Gamma","Delta","E
+
+	if(new_damage > 0 && genome_damage == 0 && my_mob) // show the biostructure if we gain some gendamage
+		to_chat(my_mob, SPAN("changeling", "You feel your genomes start to disassemble. Your special biostructure can now be easily spotted."))
+		var/obj/item/organ/internal/biostructure/biostructure = locate() in my_mob.contents
```
Точно такой же вопрос что и выше

4310V343k replied:
точно такой же ответ что и выше

4310V343k committed:
internal_organs_by_name and avoid hiding a possible runtime

4310V343k committed:
Merge remote-tracking branch 'upstream/dev' into lings!1

4310V343k commented:
🥹

NefrNerf commented on code/game/gamemodes/changeling/changeling_datum.dm:
```diff
@@ -87,17 +87,18 @@ var/global/list/possible_changeling_IDs = list("Alpha","Beta","Gamma","Delta","E
 
-	if(new_damage > 0 && genome_damage == 0 && my_mob) // show the biostructure if we gain some gendamage
+		var/obj/item/organ/internal/biostructure/biostructure = H.internal_organs_by_name[BP_CHANG]
+		biostructure.hidden = TRUE
```
А if(biostructure) решил не оставлять? Достер же говорил про такой возможный кейс, если я все правильно понял

4310V343k replied:
тогда поверим тостеру и поверим, что это intended, а ещё что хватит by_name

NefrNerf commented on code/game/gamemodes/changeling/changeling_datum.dm:
```diff
@@ -87,17 +87,18 @@ var/global/list/possible_changeling_IDs = list("Alpha","Beta","Gamma","Delta","E
+		var/obj/item/organ/internal/biostructure/biostructure = H.internal_organs_by_name[BP_CHANG]
+		biostructure.hidden = TRUE
+
+	if(new_damage > 0 && genome_damage == 0 && istype(H)) // show the biostructure if we gain some gendamage
```
Проверку на показ МБ можно вынести в отдельный метод в стиле. CheckGenDamage

4310V343k replied:
не очень представляю что он должен из себя представлять

NefrNerf reviewed (commented):
Да и еще немного логика у тебя нарушена set_genome_damage а в нем изменяется статус скрытности биоструктуры, сделай 2 функции setBiostructureHidden и setBiostructureVisible.

4310V343k committed:
brings the biostructure check back, moves hide and show to sep methods

4310V343k committed:
let's change the genome damage first, then do whatever else

4310V343k commented:
😴

4310V343k commented:
у меня на глазах слезы
